packages/angular_devkit/build_angular/test/browser/replacements_spec_large.ts

```diff
@@ -154,4 +155,53 @@ describe('Browser Builder file replacements', () => {
     );
   });
 
+  it('file replacements work with forked type checker on watch mode', async () => {
+    host.writeMultipleFiles({
+      'src/file-replaced.ts': 'export var obj = { one: 1, two: 2 };',
+      'src/file.ts': `export var obj = { one: 1 };`,
+      'src/main.ts': `
+        import { obj } from './file';
+        console.log(obj.two);
+      `,
+    });
+
+    const overrides = {
+      fileReplacements: [{
+        replace: normalize('/src/file.ts'),
+        with: normalize('/src/file-replaced.ts'),
+      }],
+      watch: true,
+    };
+
+    const unexpectedError = `Property 'two' does not exist on type '{ one: number; }'`;
+    const expectedError = `Property 'prop' does not exist on type '{}'`;
+    const logger = new TestLogger('rebuild-type-errors');
+
+    // Race between a timeout and the expected log entry.
+    const stop$ = race<null | string>(
```

Comment:
Wouldn't the below result in the same test?

```ts
    let errorAdded = false;
    runTargetSpec(host, browserTargetSpec, overrides, DefaultTimeout, logger)
      .pipe(
        tap((buildEvent) => expect(buildEvent.success).toBe(true, 'build should succeed')),
        tap(() => {
          // Introduce a known type error to detect in the logger filter.
          if (!errorAdded) {
            host.appendToFile('src/main.ts', 'console.log({}.prop);');
            errorAdded = true;
          } else {
            expect(logger).toContain(expectedError);
          }
        }),
        tap(() => {
          expect(logger).not.toContain(unexpectedError);
          logger.clear();
        }),
        take(2),
      ).toPromise();
```

Reply:
No, because the type checker logs are not synchronized with builds. Since the type checker runs separately, it will finish rebuilds separately from the main webpack process. 

In this specific case, the type checker rebuild will take longer. We need to start a build and and rebuild on the main compilation process, then wait for the forked process to send logs messages.

Reply:
Ah cause it runs in a forked process. 👍